medusa/tv/cache.py

```diff
@@ -502,50 +504,54 @@ def find_needed_episodes(self, episode, forced_search=False,
 
             cur_ep = int(cur_ep)
 
-            cur_quality = int(cur_result[b'quality'])
-            cur_release_group = cur_result[b'release_group']
-            cur_version = cur_result[b'version']
+            search_result.quality = int(cur_result[b'quality'])
+            search_result.release_group = cur_result[b'release_group']
+            search_result.version = cur_result[b'version']
 
             # if the show says we want that episode then add it to the list
-            if not show_obj.want_episode(cur_season, cur_ep, cur_quality,
+            if not show_obj.want_episode(search_result.season, cur_ep, search_result.quality,
                                          forced_search, down_cur_quality):
                 log.debug('Ignoring {0}', cur_result[b'name'])
                 continue
 
-            ep_obj = show_obj.get_episode(cur_season, cur_ep)
+            search_result.episodes = [show_obj.get_episode(search_result.season, cur_ep)]
+
+            search_result.actual_episodes = [search_result.episodes[0].episode]
+            search_result.actual_season = search_result.season
 
             # build a result object
-            title = cur_result[b'name']
-            url = cur_result[b'url']
+            search_result.name = cur_result[b'name']
+            search_result.url = cur_result[b'url']
 
             log.debug(
                 '{id}: Using cached results from {provider} for series {show_name!r} episode {ep}', {
-                    'id': ep_obj.series.indexerid,
+                    'id': search_result.episodes[0].series.indexerid,
```

Comment:
As the ep_obj was kinda misused. It was the last looped episode object instance, that was misused in the rest of the code. I made it more explicit, in adding a list of epiosde_objects to the searchr_result.episodes variable (which for every item from cache is always 1). And then just use the first one.